d2renderers/d2svg/d2svg.go

```diff
@@ -136,156 +128,175 @@ func arrowheadMarker(isTarget bool, id string, connection d2target.Connection) s
+		polylineEl.Fill = color.None
+		polylineEl.Class = "connection"
+		polylineEl.Stroke = svg_style.ConnectionTheme(connection)
+		polylineEl.Attributes = fmt.Sprintf(`stroke-width="%d"`, connection.StrokeWidth)
+
 		if isTarget {
-			path = fmt.Sprintf(`<polyline %s points="%f,%f %f,%f %f,%f"/>`,
-				attrs,
+			polylineEl.Points = fmt.Sprintf("%f,%f %f,%f %f,%f",
 				strokeWidth/2, strokeWidth/2,
 				width-strokeWidth/2, height/2,
 				strokeWidth/2, height-strokeWidth/2,
 			)
 		} else {
-			path = fmt.Sprintf(`<polyline %s points="%f,%f %f,%f %f,%f"/>`,
-				attrs,
+			polylineEl.Points = fmt.Sprintf("%f,%f %f,%f %f,%f",
 				width-strokeWidth/2, strokeWidth/2,
 				strokeWidth/2, height/2,
 				width-strokeWidth/2, height-strokeWidth/2,
 			)
 		}
+		path = polylineEl.Render()
 	case d2target.FilledDiamondArrowhead:
-		attrs := fmt.Sprintf(`class="connection" fill="%s" stroke-width="%d"`, connection.Stroke, connection.StrokeWidth)
+		polygonEl := svg_style.NewThemableElement("polygon")
+		polygonEl.Class = "connection"
+		polygonEl.Fill = svg_style.ConnectionTheme(connection)
+		polygonEl.Attributes = fmt.Sprintf(`stroke-width="%d"`, connection.StrokeWidth)
+
 		if isTarget {
-			path = fmt.Sprintf(`<polygon %s points="%f,%f %f,%f %f,%f %f,%f" />`,
-				attrs,
+			polygonEl.Points = fmt.Sprintf("%f,%f %f,%f %f,%f %f,%f",
 				0., height/2.0,
 				width/2.0, 0.,
 				width, height/2.0,
 				width/2.0, height,
 			)
 		} else {
-			path = fmt.Sprintf(`<polygon %s points="%f,%f %f,%f %f,%f %f,%f" />`,
-				attrs,
+			polygonEl.Points = fmt.Sprintf("%f,%f %f,%f %f,%f %f,%f",
 				0., height/2.0,
 				width/2.0, 0.,
 				width, height/2.0,
 				width/2.0, height,
 			)
 		}
+		path = polygonEl.Render()
 	case d2target.DiamondArrowhead:
-		attrs := fmt.Sprintf(`class="connection" fill="white" stroke="%s" stroke-width="%d"`, connection.Stroke, connection.StrokeWidth)
+		polygonEl := svg_style.NewThemableElement("polygon")
+		polygonEl.Class = "connection"
+		polygonEl.Fill = bgColor
+		polygonEl.Stroke = svg_style.ConnectionTheme(connection)
+		polygonEl.Attributes = fmt.Sprintf(`stroke-width="%d"`, connection.StrokeWidth)
+
 		if isTarget {
-			path = fmt.Sprintf(`<polygon %s points="%f,%f %f,%f %f,%f %f,%f" />`,
-				attrs,
+			polygonEl.Points = fmt.Sprintf("%f,%f %f,%f %f,%f %f,%f",
 				0., height/2.0,
 				width/2, height/8,
 				width, height/2.0,
 				width/2.0, height*0.9,
 			)
 		} else {
-			path = fmt.Sprintf(`<polygon %s points="%f,%f %f,%f %f,%f %f,%f" />`,
-				attrs,
+			polygonEl.Points = fmt.Sprintf("%f,%f %f,%f %f,%f %f,%f",
 				width/8, height/2.0,
 				width*0.6, height/8,
 				width*1.1, height/2.0,
 				width*0.6, height*7/8,
 			)
 		}
+		path = polygonEl.Render()
 	case d2target.FilledCircleArrowhead:
-		attrs := fmt.Sprintf(`class="connection" fill="%s" stroke-width="%d"`, connection.Stroke, connection.StrokeWidth)
 		radius := width / 2
+
+		circleEl := svg_style.NewThemableElement("circle")
+		circleEl.Cy = radius
```

Comment:
this should be + strokewidth/2 right?

Reply:
There is a minus sign in the original code, I only transformed it. I left a comment there saying this should be checked.